tests/unit/shader_object.cpp

```diff
@@ -0,0 +1,1357 @@
+/*
+ * Copyright (c) 2023 Nintendo
+ * Copyright (c) 2023 LunarG, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#include "../framework/layer_validation_tests.h"
+
+TEST_F(NegativeShaderObject, SpirvCodeSize) {
+    TEST_DESCRIPTION("Create shader with invalid spirv code size.");
+
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkShaderCreateInfoEXT-codeSize-08735");
+
+    AddRequiredExtensions(VK_EXT_SHADER_OBJECT_EXTENSION_NAME);
+    ASSERT_NO_FATAL_FAILURE(InitFramework());
+    auto shaderObjectFeatures = LvlInitStruct<VkPhysicalDeviceShaderObjectFeaturesEXT>();
+    auto features2 = GetPhysicalDeviceFeatures2(shaderObjectFeatures);
+    if (shaderObjectFeatures.shaderObject == VK_FALSE) {
+        GTEST_SKIP() << "shaderObject not supported.";
+    }
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &features2));
+    if (!AreRequiredExtensionsEnabled()) {
+        GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported.";
+    }
+
+    std::vector<uint32_t> spv;
+    GLSLtoSPV(&m_device->props.limits, VK_SHADER_STAGE_VERTEX_BIT, kVertexMinimalGlsl, spv);
```

Comment:
I would use the other overloaded version of `GLSLtoSPV`

```suggestion
const auto spv = GLSLToSPV(VK_SHADER_STAGE_VERTEX_BIT, kVertexMinimalGlsl);
```

for these